lib/reverse_sentence.rb

```diff
@@ -1,6 +1,46 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
+# Time complexity: O(n)
+# Space complexity: O(1)
+
+def reverse_word(string, string_start, string_end)
+  while string_start < string_end
+    temp = string[string_start]
+    string[string_start] = string[string_end]
+    string[string_end] = temp
+    string_start += 1
+    string_end -= 1
+  end
+  return string
+end
+
+def reverse_everything(string)
+  half_length = string.length / 2
+  half_length.times {|i| string[i], string[-i-1] = string[-i-1], string[i] }
 end
+
+def reverse_sentence(my_sentence)
+  return nil if my_sentence == nil
+  #reverse everything
+  reverse_everything(my_sentence)
```

Comment:
Why not just use `reverse_word(my_sentence, 0, my_sentence.length - 1)`.  Reuse one of the methods you created!